packages/@aws-cdk/aws-autoscaling-hooktargets/lib/lambda-hook.ts

```diff
@@ -0,0 +1,21 @@
+import autoscaling = require('@aws-cdk/aws-autoscaling');
+import lambda = require('@aws-cdk/aws-lambda');
+import sns = require('@aws-cdk/aws-sns');
+import { TopicHook } from './topic-hook';
+
+/**
+ * Use a Lambda Function as a hook target
+ *
+ * Internally creates a Topic to make the connection.
+ */
+export class FunctionHook implements autoscaling.ILifecycleHookTarget {
+  constructor(private readonly fn: lambda.IFunction) {
+  }
+
+  public bind(lifecycleHook: autoscaling.ILifecycleHook): autoscaling.LifecycleHookTargetProps {
```

Comment:
I wonder if `bind` should pass a `scope: Construct` (sounds like a useful pattern in general).